org-roam-db.el

```diff
@@ -367,14 +368,34 @@ connections, nil is returned."
          (files (mapcar 'car-safe (emacsql (org-roam-db) query file max-distance))))
     files))
 
+(defun org-roam-db--file-hash (&optional file-path)
+  "Compute the hash of FILE-PATH, a file or current buffer."
+  (let* ((file-p (and file-path))
+         (file-path (or file-path
+                        (buffer-file-name (current-buffer))))
+         (orig-buffer (current-buffer))
+         (encrypted-p (and file-path
+                           (string= (org-roam--file-name-extension file-path)
+                                    "gpg"))))
+    (if (and encrypted-p file-p)
+        (with-temp-buffer
+          (set-buffer-multibyte nil)
+          (insert-file-contents-literally file-path)
+          (secure-hash 'sha1 (current-buffer)))
+      (with-temp-buffer
+        (if file-p
+            (insert-file-contents file-path)
+          (insert-buffer-substring orig-buffer))
```

Comment:
If the `file-path` option is not passed, it's probably cheaper to just call `secure-hash` on the current buffer, without creating a new temp buffer and inserting the exact same contents.

Reply:
Yes, sounds reasonable.  Created a new patch.